addons/account/models/account_bank_statement.py

```diff
@@ -36,23 +37,61 @@ class AccountBankStmtCashWizard(models.Model):
     _name = 'account.bank.statement.cashbox'
     _description = 'Bank Statement Cashbox'
 
+    def _default_currency(self):
+        return self.env.company.currency_id.id
```

Comment:
taking the currency_id from self.env is not always true. I think what we should do is having a computed field (no need to store it as it is just for the view)

currency_id = fields.Many2one('res.currency', compute='_compute_currency')

in the compute method, you put depends on 'start_bank_stmt_ids' and 'end_bank_stmt_ids', an take the start_bank_stmt_ids[0].currency_id ou celui du end en fonction duquel est présent.

Dans l'override du POS, tu rajoute un depends sur les pos_config_ids et dans l'override tu prends la currency de la config (il n'y pas de statements dans ce cas normalement.